src/main/java/org/candlepin/subscriptions/security/SecurityConfig.java

```diff
@@ -114,6 +132,20 @@ public AuthenticationEntryPoint restAuthenticationEntryPoint() {
         return new RestAuthenticationEntryPoint(new IdentityHeaderAuthenticationFailureHandler(mapper));
     }
 
+    @Bean
+    public FilterRegistrationBean<OptInFilter> optInFilterRegistration(OptInController optInController) {
```

Comment:
Do we want to register this under the Spring Security filter chain similarly to `identityHeaderAuthenticationFilter`?

Reply:
I originally attempted to do this, but couldn't seem to get it to filter based on URL unless it was moved out the security chain. Not sure why, but I'll make another attempt.

Reply:
@awood I made another attempt at doing this, but failed. I can't seem to get the url filters to get applied. Not sure why. If this is something that you'd really like to see, let's create a card to address it so that we can get this merged.